include/pisa/topk_queue.hpp

```diff
@@ -10,7 +10,8 @@ using Threshold = float;
 struct topk_queue {
     using entry_type = std::pair<float, uint64_t>;
 
-    explicit topk_queue(uint64_t k) : m_threshold(0), m_k(k) { m_q.reserve(m_k + 1); }
+    explicit topk_queue(uint64_t k) : m_threshold(0), m_initialized_threshold(0), m_k(k) { m_q.reserve(m_k + 1); }
+    explicit topk_queue(uint64_t k, Threshold t) : m_threshold(0), m_initialized_threshold(t), m_k(k) { m_q.reserve(m_k + 1); }
```

Comment:
We can now remove the zero-initializers (`m_threshold` in both and `m_initialized_threshold` in the first constructor) now that we set it below.

Reply:
Yes, I just noticed this - fixing a couple of other issues too. Thanks!